Source/BSN.Commons/Infrastructure/MessageBroker/Jms/JmsEventAggregator.cs

```diff
@@ -0,0 +1,122 @@
+using BSN.Commons.Infrastructure.MessageBroker.EventAggregator;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.Contracts;
+using BSN.Commons.Infrastructure.MessageBroker.EventContracts.EventAggregator.EventModels;
+using Apache.NMS;
+using Apache.NMS.ActiveMQ;
+using Apache.NMS.Util;
+using Newtonsoft.Json;
+
+namespace BSN.Commons.Infrastructure.MessageBroker.Jms
+{
+    /// <summary>
+    /// Represents an implementation of <see cref="IEventAggregator"/>.
+    /// using JMS as the underlying message broker.
+    /// </summary>
+    public class JmsEventAggregator : IEventAggregator
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="JmsEventAggregator"/> class.
+        /// </summary>
+        public JmsEventAggregator(
+            JmsConnectionOptions jmsConnectionOptions, 
+            IEventAggregatorSubscriptionManager eventAggregatorSubscriptionManager
+            )
+        {
+            _subscriptionManager = eventAggregatorSubscriptionManager ?? new InMemoryEventAggregatorSubscriptionManager();
+
+            IConnectionFactory factory = new ConnectionFactory(jmsConnectionOptions.BrokerUri);
+            _connection = factory.CreateConnection();
+            _connection.Start();
+            _session = _connection.CreateSession();
+            _destination = SessionUtil.GetDestination(_session, jmsConnectionOptions.QueueName);
+            
+            _producer = _session.CreateProducer(_destination);
+            _consumers = new Dictionary<string, IMessageConsumer>();
+        }
+        
+        /// <inheritdoc />
+        public void Publish<TEventModel>(IEvent<TEventModel> @event) where TEventModel : IEventDataModel
+        {
+            string serializedEvent = JsonConvert.SerializeObject(@event.DataModel);
+            
+            ITextMessage message = _session.CreateTextMessage(serializedEvent);
+            
+            _producer.Send(message);
+        }
+
+        /// <inheritdoc />
+        public Task PublishAsync<TEventModel>(IEvent<TEventModel> @event) where TEventModel : IEventDataModel
+        {
+            return Task.Run(() => Publish(@event));
+        }
+
+        /// <inheritdoc />
+        public bool HasSubscriptionForEvent<TEvent, TEventDataModel>() where TEvent : IEvent<TEventDataModel> where TEventDataModel : IEventDataModel
+        {
+            return _subscriptionManager.HasSubscriptionsForEvent<TEvent>();
+        }
+
+        /// <inheritdoc />
+        public void Subscribe<TEvent, TEventDataModel>(IEventReceiver eventReceiver) where TEvent : IEvent<TEventDataModel> where TEventDataModel : IEventDataModel
+        {
+            string eventName = typeof(TEvent).Name;
+            
+            _subscriptionManager.AddSubscription<TEvent>(eventReceiver);
+            
+            var consumer = _session.CreateConsumer(_destination);
```

Comment:
What is behavior, if call subscribe with same input twice?

Reply:
changed it a bit
now it does not add a new consumer for same eventName
but adds new Listener for new eventRecievers